### PR TITLE
[nrfconnect] Added nRF Connect platform overview guide.

### PR DESCRIPTION
**Pull request by kkasperczyk-no**

### CODE REVIEW

File: docs/guides/nrfconnect_platform_overview.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+The nRF Connect SDK is based around Zephyr RTOS, which is scalable real-time operating system, dedicated for the connected, resource constrained devices. Zephyr supports multiple hardware platforms, provides hardware drivers, application protocols, protocol stacks and even more. Besides of the Zephyr, nRF Connect SDK also intergrates other projects like crypto library mbed TLS, MCU Bootloader or Thread stack implementation [OpenThread](https://openthread.io/).
+
+The diagram presented below, shows simplified structure of the CHIP application running on the nRF Connect platform and using BLE and Thread stacks for communication purposes: 
+
+![nrfconnect overview](../../examples/platform/nrfconnect/doc/images/nrfconnect_overview.png)
```
Comment (LuDuda) [inline docs/guides/nrfconnect_platform_overview.md:11]: Image looks much better, thanks! 👍 

Small nits (sorry 😄 ):

- Resize IPv6 block to be under both UDP and TCP - full width
- There is a larger gap between Thread Stack and IP, and Thread start and Zephyr integration (visualization issue)
- You will kill me, but border around CHIP stack and Zephyr integration.. is very strong. Can we do sth more smooth and lightweight? 👍

File: docs/guides/nrfconnect_platform_overview.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+> **Note**: The diagram doesn't show all projects components, but only the most important for the typical CHIP application, because of the readability reasons.
+
+<hr>
+
+### BLE and Thread stacks
+
+In the nRF Connect platform applications BLE interface is used to perform pairing and Thread network provisioning operations between CHIP device and CHIP controller. After that fully provisioned device is able to communicate with other devices inside of the Thread network.
+
+For the BLE communication purposes application is using BLE stack, in which BLE host part is provided by the Zephyr RTOS and the BLE controller is implemented in the nRF Connect SDK's driver.
```
Comment (LuDuda) [inline docs/guides/nrfconnect_platform_overview.md:21]: Can we add reference that BLE controller is "SoftDevice Controller" and add a link: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrfxlib/softdevice_controller/README.html

**Review by LuDuda — COMMENTED** [review]
Kamil this is really great! Thank you for this work.

File: docs/guides/nrfconnect_platform_overview.md (hunk 1 above)
Comment (kkasperczyk-no) [reply docs/guides/nrfconnect_platform_overview.md:11]: I working on figuring out how to fetch and merge your nits into the .png image using git :smile: If it comes to the border, do you mean only its thickness or color is also too strong?

File: docs/guides/nrfconnect_platform_overview.md (hunk 2 above)
Comment (kkasperczyk-no) [reply docs/guides/nrfconnect_platform_overview.md]: Done

**Review by LuDuda — APPROVED** [review]
Well done!

**Review by greg-fer — APPROVED** [review]

**Review by Damian-Nordic — APPROVED** [review]

**Pull request closed without merging**